build-system/tasks/report-test-results.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview This file implements the `gulp report-test-results` task, which [[does something]]
+ */
+
+'use strict';
+
+const fetch = require('node-fetch');
+const fs = require('fs').promises;
+const log = require('fancy-log');
+const path = require('path');
+const {
+  travisBuildNumber,
+  travisJobNumber,
+  travisCommitSha,
+} = require('../common/travis');
+const {cyan, green, yellow} = require('ansi-colors');
+
+const REPORTING_API_URL = 'https://amp-test-cases.appspot.com/report';
+
+async function getReport(testType) {
+  const report = await fs
+    .readFile(`result-reports/${testType}.json`)
+    .then(JSON.parse);
+
+  return addJobAndBuildInfo(testType, report);
+}
+
+function addJobAndBuildInfo(testType, reportJson) {
+  const build = {
+    buildNumber: travisBuildNumber(),
```

Comment:
Should we throw an error if there isn't a travis build number (e.g. if this is called outside a travis build)?

Reply:
WDYT?

Reply:
I think yes; I don't believe this should be getting called outside of builds at all.

Reply:
Added exceptions for when any of the `travisSomethingSomething` calls fail, let me know what you think.